src/cargo/ops/fix.rs

```diff
@@ -274,6 +317,46 @@ fn migrate_manifests(ws: &Workspace<'_>, pkgs: &[&Package]) -> CargoResult<()> {
     Ok(())
 }
 
+fn rename_dep_fields_2024(parent: &mut dyn toml_edit::TableLike, dep_kind: &str) -> usize {
+    let mut fixes = 0;
+    for target in parent
+        .get_mut(dep_kind)
+        .and_then(|t| t.as_table_like_mut())
+        .iter_mut()
+        .flat_map(|t| t.iter_mut())
+        .filter_map(|(_k, t)| t.as_table_like_mut())
+    {
+        if rename_table(target, "default_features", "default-features") {
+            fixes += 1;
+        }
+    }
+    fixes
+}
+
+fn rename_array_of_target_fields_2024(root: &mut dyn toml_edit::TableLike, kind: &str) -> usize {
+    let mut fixes = 0;
+    for target in root
+        .get_mut(kind)
+        .and_then(|t| t.as_array_of_tables_mut())
+        .iter_mut()
+        .flat_map(|t| t.iter_mut())
+    {
+        fixes += rename_target_fields_2024(target);
+    }
+    fixes
+}
+
+fn rename_target_fields_2024(target: &mut dyn toml_edit::TableLike) -> usize {
+    let mut fixes = 0;
+    if rename_table(target, "crate_type", "crate-type") {
+        fixes += 1;
+    }
+    if rename_table(target, "proc_macro", "proc-macro") {
+        fixes += 1;
+    }
+    fixes
+}
+
 fn rename_table(parent: &mut dyn toml_edit::TableLike, old: &str, new: &str) -> bool {
```

Comment:
Should we change `rename_table` to return a number so that we can remove some `if` branches?
(and a bit more visually consistent?)

Reply:
Had considered it but was originally holding off.  Your commenting pushed me over the edge.